### PR TITLE
Adding conda-vendor package

### PR DESCRIPTION
**Pull request by tylerpotts**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by ghoersti** [comment]
Hello, I am a maintainer of conda-vendor :)

**Comment by tylerpotts** [comment]
@conda-forge/help-python Ready for your review plz

File: recipes/conda-vendor/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+source:
+  url: https://github.com/MetroStar/conda-vendor/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 2e6db8144ff88f07dcbd0c5f9381d3dd348d3034f3feced895df341373e03081
+
+build:
+  skip: True  # [win]
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
```
Comment (goanpeca) [inline recipes/conda-vendor/meta.yaml:14]: I think 

`script: "{{ PYTHON }} -m pip install . -vv"`

Should be enough, following on the examples https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml

Comment (tylerpotts) [reply recipes/conda-vendor/meta.yaml]: thanks! modified per your recommendation

File: recipes/conda-vendor/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - pip
+    - python
+    - conda-lock
+  run:
+    - python
+    - conda-lock
```
Comment (dopplershift) [inline recipes/conda-vendor/meta.yaml:23]: Upstream also lists `pyyaml`:
```suggestion
    - conda-lock
    - pyyaml
```

File: recipes/conda-vendor/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+build:
+  skip: True  # [win]
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - pip
+    - python
+    - conda-lock
```
Comment (dopplershift) [inline recipes/conda-vendor/meta.yaml:20]: Doesn't look like you need this at package build time:
```suggestion
```

File: recipes/conda-vendor/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+source:
+  url: https://github.com/MetroStar/conda-vendor/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: a9d99fb2bd5b815bbebc8628242debbcbff5f6a1948624e720a95ee91a9a4cc9
+
+build:
+  skip: True  # [win]
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
```
Comment (dopplershift) [inline recipes/conda-vendor/meta.yaml:14]: The package is pure Python, so it should be able to be noarch python, provided the console script is properly listed here (to fix the windows build):
```suggestion
  number: 0
  script: "{{ PYTHON }} -m pip install . -vv"
  noarch: python
  entry_points:
    - conda-vendor = conda_vendor.__main__:cli
```

File: recipes/conda-vendor/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+requirements:
+  host:
+    - pip
+    - python
```
Comment (dopplershift) [inline recipes/conda-vendor/meta.yaml:19]: If going noarch:
```suggestion
    - python >=3.6
```

File: recipes/conda-vendor/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    - conda-lock
+  run:
+    - python
```
Comment (dopplershift) [inline recipes/conda-vendor/meta.yaml:22]: For noarch:
```suggestion
    - python >=3.6
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
Would be good to make this noarch if we can (it drastically cuts down the number of builds). @ghoersti Can you confirm you are willing to be listed as a maintainer for this recipe?

**Comment by tylerpotts** [comment]
@dopplershift Thanks for the review! @ghoersti Indicated his willingness earlier in this thread: https://github.com/conda-forge/staged-recipes/pull/16125#issuecomment-916442421

When you say:
> provided the console script is properly listed here

I believe the console script `{{ PYTHON }} -m pip install . -vv` is correct for windows. Is there something in particular you're looking for here?

**Comment by dopplershift** [comment]
I was actually referring to the `conda-vendor = conda_vendor.__main__:cli` line which you have already incorporated.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**